### DISCUSSION TITLE
Need help to use fflate for decompressing a local database asynchronously

### DISCUSSION THREAD

**Discussion by Sbrjt** [101arrowz/fflate#215] [Q&A] [answered]
I'm using a compressed sql.js database in my website that I want to decompress. I was using pako (with a cdn) to do so, but it's blocking my main thread, making the site unresponsive for a few secs.

	const response = await fetch('data.db.gz')
	const buffer = await response.arrayBuffer()
	const decompressedBuffer = pako.inflate(buffer)
	const db = new SQL.Database(new Uint8Array(decompressedBuffer))
	
Can you tell me how to do so using fflate asynchronously? And where can I find the inflate-only cdn for fflate?

I found this [SO answer](https://stackoverflow.com/a/64993991/14313852) but couldn't figure out how to do it for my database.

**Comment by 101arrowz** [comment, marked as answer]
There isn't really an inflate-only CDN. I've been waiting for a while for a CDN service to offer the ability to tree-shake based on the function names specified in a query parameter or something to offer exactly that kind of functionality, but unfortunately I don't think it exists; you'll probably have to use the 11kB minzipped CDN bundle. You can do that by adding the HTML provided in the README to your page:

```html
<script src="https://cdn.jsdelivr.net/npm/fflate@0.8.2/umd/index.js"></script>
```

Beyond that, to make your above code use `fflate` and not hang the page, all you'd change is the `decompressedBuffer` line.

```js
const decompressedBuffer = await new Promise((resolve, reject) => {
  fflate.inflate(buffer, (err, result) => err ? reject(err) : resolve(result));
});
```
